larrecodnn/ImagePatternAlgs/Modules/RawWaveformClnSigDump_module.cc

```diff
@@ -0,0 +1,807 @@
+    isSorted = true;
+  }
+  void
+  add(const int& track_id, const std::string& gname)
+  {
+    this->track_id_map.push_back(std::make_pair(track_id, gname));
+    generator_names.emplace(gname);
+    isSorted = false;
+  }
+  bool
+  has_gen(std::string gname)
+  {
+    return static_cast<bool>(generator_names.count(gname));
+  };
+  std::string
+  get_gen(int tid)
+  {
+    if (!isSorted) { this->sort_now(); }
+    return std::lower_bound(track_id_map.begin(),
+                            track_id_map.end(),
+                            tid,
+                            [](const auto& a, const auto& b) { return (a.first < b); })
+      ->second;
+  };
+};
+
+//-----------------------------------------------------------------------
+nnet::RawWaveformClnSigDump::RawWaveformClnSigDump(fhicl::ParameterSet const& p)
+  : EDAnalyzer{p}
+  , fDumpWaveformsFileName(p.get<std::string>("DumpWaveformsFileName", "dumpwaveforms"))
+  , fDumpCleanSignalFileName(p.get<std::string>("CleanSignalFileName", "dumpcleansignal"))
+  , fSimulationProducerLabel(p.get<std::string>("SimulationProducerLabel", "larg4Main"))
+  , fSimChannelLabel(p.get<std::string>("SimChannelLabel", "elecDrift"))
+  , fDigitModuleLabel(p.get<std::string>("DigitModuleLabel", "simWire"))
+  , fCleanSignalDigitModuleLabel(p.get<std::string>("CleanSignalDigitModuleLabel", "simWire:signal"))
+  , fUseFullWaveform(p.get<bool>("UseFullWaveform", true))
+  , fShortWaveformSize(p.get<unsigned int>("ShortWaveformSize"))
+  , fEstIndFWForOffset(p.get<int>("EstIndFWForOffset",14))
+  , fEstColFWForOffset(p.get<int>("EstIndFWForOffset",32))
+  , fSelectGenLabel(p.get<std::string>("SelectGenLabel", "ANY"))
+  , fSelectProcID(p.get<std::string>("SelectProcID", "ANY"))
+  , fSelectPDGCode(p.get<int>("SelectPDGCode", 0))
+  , fPlaneToDump(p.get<std::string>("PlaneToDump", "U"))
+  , fMinParticleEnergyGeV(p.get<double>("MinParticleEnergyGeV", 0.))
+  , fMinEnergyDepositedMeV(p.get<double>("MinEnergyDepositedMeV", 0.))
+  , fMinNumberOfElectrons(p.get<int>("MinNumberOfElectrons", 1000))
+  , fMaxNumberOfElectrons(p.get<int>("MaxNumberOfElectrons", 100000))
+  , fMinPureSignalADCs(p.get<int>("MinPureSignalADCs", 0))
+  , fSaveSignal(p.get<bool>("SaveSignal", true))
+  , fMaxNoiseChannelsPerEvent(p.get<int>("MaxNoiseChannelsPerEvent", 1000))
+  , fCollectionPlaneLabel(p.get<std::string>("CollectionPlaneLabel", "Z"))
+{
+  if (std::getenv("CLUSTER") && std::getenv("PROCESS")) {
+    fDumpWaveformsFileName += string(std::getenv("CLUSTER")) + "-" + string(std::getenv("PROCESS")) + "-";
+    fDumpCleanSignalFileName += string(std::getenv("CLUSTER")) + "-" + string(std::getenv("PROCESS")) + "-";
+  }
+
+  if (fDigitModuleLabel.empty() && fCleanSignalDigitModuleLabel.empty()) {
+    throw cet::exception("RawWaveformClnSigDump")
+      << "Both DigitModuleLabel and CleanSignalModuleLabel are empty";
+  }
+  fRand = new TRandom3(0);
+  fRand->SetSeed(0);
```

Comment:
Should the seed be configurable (in this module and in the other one)?  Also, any reason to not use the CLHEP-based `createEngine(...)` functionality provided by art?

Reply:
No, the seed doesn't need to be configurable.  What is important is for each job to start with a unique (pseudo-) random sequence every time.